cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java

```diff
@@ -21,9 +28,50 @@
  */
 public class SlangBootstrap {
 
+    private static final String USER_CONFIG_DIR = "configuration";
+    private static final String USER_CONFIG_FILENAME = "cslang.properties";
+    private static final String USER_CONFIG_FILEPATH = USER_CONFIG_DIR + File.separator + USER_CONFIG_FILENAME;
+    private static final String SUBSTITUTION_REGEX = "\\$\\{([^${}]+)\\}"; // ${VAR}
+    private static final Pattern SUBSTITUTION_PATTERN = Pattern.compile(SUBSTITUTION_REGEX);
+
     public static void main(String[] args) throws IOException {
+        loadUserProperties();
         System.out.println("Loading..");
         Bootstrap.main(args);
     }
 
+    private static void loadUserProperties() throws IOException {
+        String appHome = System.getProperty("app.home");
+        String propertyFilePath = appHome + File.separator + USER_CONFIG_FILEPATH;
+        File propertyFile = new File(propertyFilePath);
+        Properties rawProperties = new Properties();
+        if (propertyFile.exists()) {
+            try (InputStream propertiesStream = new FileInputStream(propertyFilePath)) {
+                rawProperties.load(propertiesStream);
+            }
+        }
+        Enumeration<?> e = rawProperties.propertyNames();
+        while (e.hasMoreElements()) {
+            String key = (String) e.nextElement();
+            String value = rawProperties.getProperty(key);
+            value = substitutePropertyReferences(value);
+            System.setProperty(key, value);
+        }
+    }
+
+    private static String substitutePropertyReferences(String value) {
+        Matcher mather = SUBSTITUTION_PATTERN.matcher(value);
+        Set<String> variableNames = new HashSet<>();
+        while (mather.find()) {
+            variableNames.add(mather.group(1));
+        }
```

Comment:
maybe split this method into two methods at this line, the above part "findSystemPropertiesNames", the below part "replaceSystemPropertiesNamesWithValues" or something similar. WDYT?